src/avtt/content-script.js

```diff
@@ -0,0 +1,580 @@
+        }
+    });
+}
+
+function getName(request) {
+    return request.name != "" ? request.name : request.character.name;
+}
+
+function subRolls(text, overrideCB = null) {
+    let replaceCB = overrideCB;
+    
+    if (!overrideCB) {
+        replaceCB = (dice, modifier) => {
+            return dice == "" ? modifier : `!!(${dice}${modifier})`;
+        }
+    }
+
+    const result = replaceRolls(text, replaceCB);
+    return result.replace(/\]\](\s*\+\s*)\[\[/g, '$1')
+}
+
+function parseDescription(request, description, {
+    bulletList = true,
+    notes = true,
+    tables = true,
+    bolded = true
+} = {}) {
+    // Trim lines
+    description = description.split('\n').map(s => s.trim()).join('\n');
+
+    if (!settings["subst-vtt"])
+        return description;
+        
+    notes && (description = formatNotesInDescription(description));
+    tables && (description = formatTableInDescription(description));
+    bulletList ? (description = formatBulletListsInDescription(description)) : (description = formatSeparateParagraphsInDescription(description));
+    bolded && (description = formatBoldedSectionInDescription(description));
+    const replaceCB = (dice, modifier) => {
+        return dice == "" ? modifier : `${dice}${modifier} (!!(${dice}${modifier}))`;
+    }
+    return subRolls(description, replaceCB);
+}
+
+function subDamageRolls(text) {
+    return subRolls(text, (dice, modifier) => {
+        const prefix = settings["auto-roll-damage"] ? "!(" : "!!(";
+        modifier = settings['critical-homebrew'] == CriticalRules.HOMEBREW_DOUBLE ? `)${modifier}` : `${modifier})`;
+        return dice == "" ? modifier : `${prefix}${dice}${modifier}`;
+    });
+}
+
+function damagesToRolls(damages, damage_types, crits, crit_types) {
+    return [
+        ...damages.map((value, index) => [damage_types[index], subDamageRolls(value)]),
+        ...(settings['critical-homebrew'] == CriticalRules.HOMEBREW_DOUBLE ? 
+            [] : 
+            crits.map((value, index) => ["Crit " + crit_types[index], subDamageRolls(value)])
+        )
+    ];
+}
+
+function template(rolls, col1 = "Name", col2 = "Roll") {
+    return `| ${col1} | ${col2} |
+| :--- | :--- |
+${rolls.map(([key, value]) => {
+    return `| ${key} | ${value} |`
+}).join('\n')}    
+
+`;
+}
+
+function formatTableInDescription(description) {
+    const matchTable =  new RegExp(/(^\n\n([^\n\r]+)\n([^\n\r]+)\n\n\n\n$)(^\n([^\n\r]+)\n([^\n\r]+)\n\n$)+/gm);
+    const matchHead =  new RegExp(/(^\n\n([^\n\r]+)\n([^\n\r]+)\n\n\n\n$)/gm);
+    const matchRow =  new RegExp(/(^\n([^\n\r]+)\n([^\n\r]+)\n\n$)/gm);
+
+    return description.replace(matchTable, (match) => {
+        let [_1, _2, firstHead, secondHead] = match.matchAll(matchHead).next().value;
+        
+        // Skip first as it is matched by the matchHead too
+        const [_, ...rows] = [...match.matchAll(matchRow)].map(([_1, _2, firstRow, secondRow]) => {
```

Comment:
I really hate the ugliness of using spread operators everywhere. Please prioritize readability and maintainability vs one-liners.
```js
const allMatches = reMatchAll(match, matchRow];
// Skip first as it is matched by the matchHead too
const rows = allMatch.map(match => `| ${match[2]} | ${match[3]} |`).slice(1);
```